WebRTC 1:n 접속을 위한 simple peer 로직 적용

voidmatcha opened the pull request:
> resolve #27
- [x] simple peer 연결을 위한 소켓 mock server 추가
- [x] simple peer 모듈 등 설치
- [x] 1:n 연결을 위한 client 환경 구성

Commit:
feat(webrtc): WebRTC mock server 추가(#27)

- 아래 reference를 바탕으로 구성함
- disconnect시 peer 목록을 갱신할 수 있도록 하는 코드를 추가해야함
https://github.com/coding-with-chaim/group-video-final/blob/master/server.js

Commit:
feat(webrtc): simple-peer등 필요한 모듈 설치 (#27)

- react-router-prop-types
- simple-peer
- socket.io-client
- uuid
등의 모듈 설치

Commit:
feat(webrtc): simple-peer 1:n을 연결을 위한 client 설정 (#27)

- ConferenctButton 컴포넌트 생성
- ConferenceRoom 페이지 생성
- PeerVideo 컴포넌트 생성
- recat-router-dom Route 구성 수정

SkynI25 commented:
구현하시느라 정말 수고많으셨습니다..전체적으로 확인한 다음 코드 리뷰 남기겠습니다 :)

SkynI25 commented on src/pages/ConferenceRoom/ConferenceRoom.js:
```diff
@@ -0,0 +1,135 @@
+  function addPeer(incomingSignal, callerID, stream) {
+    const peer = new Peer({
+      initiator: false,
+      trickle: false,
```
trickle 옵션과 관련해서 doc를 읽고 있는데 `뭥미?` 스럽군요..(한마디로 어렵;;) 이 부분을 false로 한 이유가 있을까요?

https://webrtchacks.com/trickle-ice/

voidmatcha replied:
https://cloud.google.com/solutions/gpu-accelerated-streaming-using-webrtc?hl=ko#connectivity_tracing

해당 옵션에 대해서 제대로 알지 못했었는데 true로 바꾸는게 더 좋을 것 같습니다. 반영하겠습니다.

SkynI25 commented on src/pages/ConferenceRoom/ConferenceRoom.js:
```diff
@@ -0,0 +1,135 @@
+  }
+
+  useEffect(() => {
+    socketRef.current = io.connect('localhost:8000');
```
혹시 socket 인스턴스를 일반 변수가 아닌 ref 훅에 초기화한 이유가 있을까요? 스코프와 관련이 있나요?

voidmatcha replied:
useRef를 사용해야 값이 변하더라도 re-rendering이 일어나지 않는다고 알고있습니다.

SkynI25 commented on src/pages/ConferenceRoom/ConferenceRoom.js:
```diff
@@ -0,0 +1,135 @@
+
+  return (
+    <Container>
+      <StyledVideo muted ref={userVideo} autoPlay playsInline />
```
`muted` 설정이 되있는데 실제 실행을 시켜보면 제대로 적용이 안되는거같아요..
![image](https://user-images.githubusercontent.com/20556561/97021591-fdf20200-158d-11eb-82a1-afafedc6f997.png)

`PeerVideo` 는 다른 유저가 들어왔을때 생기는 비디오 태그이고 `StyledVideo` 컴포넌트는 자기 자신에 대한 비디오 태그죠?

voidmatcha replied:
네 맞습니다. muted옵션에 대해서는 좀 더 확인해보도록 하겠습니다.

SkynI25 reviewed (commented):
![image](https://user-images.githubusercontent.com/20556561/97021974-84a6df00-158e-11eb-90d4-0c01273b4f91.png)

prop 타입이 유효하지 않고 key 에 대한 설정이 필요하다는 warning이 있네요, 수정하면 좋을 것 같습니다! :)

Commit:
refactor: 리뷰받은 내용 반영 (#30)
- trickle 옵션 true로 변경
- PeerVideo video 옵션 muted 추가
- PeerVideo의 peer PropTypes을 object.isRequired로 변경
- .eslintrc의 react/forbid-prop-types rules 추가
- peers의 map index를 PeerVideo 컴포넌트의 key로 변경경

voidmatcha commented:
> ![image](https://user-images.githubusercontent.com/20556561/97021974-84a6df00-158e-11eb-90d4-0c01273b4f91.png)
> 
> prop 타입이 유효하지 않고 key 에 대한 설정이 필요하다는 warning이 있네요, 수정하면 좋을 것 같습니다! :)

반영하였습니다.

SkynI25 reviewed (approved):
확인했습니다 :)

Commit:
Merge branch 'master' into feat/#27
